__tests__/app.test.js

```diff
@@ -135,47 +137,149 @@ describe("GET /api/articles", () => {
+      });
+  });
+});
+
+describe("POST /api/articles/:article_id/comments", () => {
+  test("201: Responds with 201 code and the posted comment", () => {
+    const newComment = {
+      username: "rogersop",
+      body: "AHHHHHHHH test banana grapefruit AHHHH!!!",
+    };
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(newComment)
+      .expect(201)
+      .then((res) => {
+        expect(res.body.comment).toEqual(
+          expect.objectContaining({
+            body: "AHHHHHHHH test banana grapefruit AHHHH!!!",
+            votes: 0,
+            author: "rogersop",
+            article_id: 1,
+            created_at: expect.any(String),
+          })
+        );
+      });
+  });
+
+  test("404: Responds with 404 when user not found", () => {
+    const newComment = {
+      username: "jimbob",
+      body: "test",
+    };
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(newComment)
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("404 - Not found, user doesn't exist!");
+      });
+  });
+
+  test("404: Responds with a 404 when article not found", () => {
+    const newComment = {
+      username: "rogersop",
+      body: "AHHHHHHHH test banana grapefruit AHHHH!!!",
+    };
+    const fakeArticleId = 99999;
+    return request(app)
+      .post(`/api/articles/${fakeArticleId}/comments`)
+      .send(newComment)
+      .expect(404)
+      .then((res) => {
+        expect(res.body.msg).toBe(
+          `404 - No article found for article_id ${fakeArticleId}`
+        );
+      });
+  });
+
+  test("400: Responds with a 400 when banana/string input as ID", () => {
+    const newComment = {
+      username: "rogersop",
+      body: "AHHHHHHHH test banana grapefruit AHHHH!!!",
+    };
+    const fakeArticleId = "banana";
+    return request(app)
+      .post(`/api/articles/${fakeArticleId}/comments`)
+      .send(newComment)
+      .expect(400)
+      .then((res) => {
+        expect(res.body.msg).toBe(`400 - Bad request, invalid type`);
+      });
+  });
+
+  test("400: Responds with 400 when no BODY entered", () => {
+    const newComment = {
+      username: "rogersop",
+      body: "",
+    };
+    return request(app)
+      .post(`/api/articles/1/comments`)
+      .send(newComment)
+      .expect(400)
+      .then((res) => {
+        expect(res.body.msg).toBe(
+          `400 - Bad request, you haven't typed a comment!`
+        );
+      });
+  });
+
+  test("400: Responds with a 400 when body is all white spaces", () => {
+    const newComment = {
+      username: "rogersop",
+      body: "                                          ",
+    };
+    return request(app)
+      .post(`/api/articles/1/comments`)
+      .send(newComment)
+      .expect(400)
+      .then((res) => {
+        expect(res.body.msg).toBe(
+          `400 - Bad request, you haven't typed a comment!`
+        );
+      });
+  });
 });
```

Comment:
Some extra tests to consider would be:

What should happen when the newComment doesnt not have a body property at all?
What should happen when the newComment has extra key value pairs?